src/test/unit_tests/jest_tests/test_nc_lifecycle_cli.test.js

```diff
@@ -0,0 +1,99 @@
+const { TMP_PATH, set_nc_config_dir_in_config, TEST_TIMEOUT, exec_manage_cli } = require('../../system_tests/test_utils');
+const BucketSpaceFS = require('../../../sdk/bucketspace_fs');
+const { TYPES, ACTIONS } = require('../../../manage_nsfs/manage_nsfs_constants');
+const NamespaceFS = require('../../../sdk/namespace_fs');
+const endpoint_stats_collector = require('../../../sdk/endpoint_stats_collector');
+const os_utils = require('../../../util/os_utils');
+
+const new_umask = process.env.NOOBAA_ENDPOINT_UMASK || 0o000;
+const old_umask = process.umask(new_umask);
+console.log('test_nc_lifecycle_cli: replacing old umask: ', old_umask.toString(8), 'with new umask: ', new_umask.toString(8));
+
+const config_root = path.join(TMP_PATH, 'config_root_nc_lifecycle');
+const root_path = path.join(TMP_PATH, 'root_path_nc_lifecycle/');
+const config_fs = new ConfigFS(config_root);
+
+function make_dummy_object_sdk(account_json) {
+    return {
+        requesting_account: account_json
+    };
+}
+
+
+describe('noobaa cli - lifecycle', () => {
+    const bucketspace_fs = new BucketSpaceFS({ config_root }, undefined);
+    const test_bucket = 'test-bucket';
+    const test_bucket2 = 'test-bucket2';
+    const test_bucket_path = `${root_path}/${test_bucket}`;
+    const test_key1 = "test_key1";
+    const account_options1 = {uid: 2002, gid: 2002, new_buckets_path: root_path, name: 'user2', config_root, allow_bucket_creation: "true"};
+    const ns_src = new NamespaceFS({
+        bucket_path: test_bucket_path,
+        bucket_id: '1',
+        namespace_resource_id: undefined,
+        access_mode: undefined,
+        versioning: undefined,
+        force_md5_etag: false,
+        stats: endpoint_stats_collector.instance(),
+    });
+
+    beforeAll(async () => {
+        await fs_utils.create_fresh_path(config_root, 0o777);
+        set_nc_config_dir_in_config(config_root);
+        await fs_utils.create_fresh_path(root_path, 0o777);
+        const res = await exec_manage_cli(TYPES.ACCOUNT, ACTIONS.ADD, account_options1);
+        const json_account = JSON.parse(res).response.reply;
+        console.log(json_account);
+        const dummy_sdk = make_dummy_object_sdk(json_account);
+        await bucketspace_fs.create_bucket({ name: test_bucket }, dummy_sdk);
+        await bucketspace_fs.create_bucket({ name: test_bucket2 }, dummy_sdk);
+    });
+
+    afterEach(async () => {
+        await bucketspace_fs.delete_bucket_lifecycle({ name: test_bucket });
+        await bucketspace_fs.delete_bucket_lifecycle({ name: test_bucket2 });
+    });
+
+    afterAll(async () => {
+        await fs_utils.folder_delete(`${root_path}/${test_bucket}`);
+        await fs_utils.folder_delete(`${root_path}/${test_bucket2}`);
+        await fs_utils.folder_delete(root_path);
+        await fs_utils.folder_delete(config_root);
+    }, TEST_TIMEOUT);
+
+    it('lifecycle_cli - abort mpu by number of days ', async () => {
```

Comment:
can we do have a similar test that has both existing prefix and days_after_initiation?
and in the file system we will have one upload key that pass the filter and one that does not?